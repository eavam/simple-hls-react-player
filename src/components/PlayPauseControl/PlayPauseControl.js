// @flow
import React from 'react';
import styled from 'react-emotion';

import withConnect from '../../hocs/withConnect';
import { togglePause } from '../../actionTypes';
import { baseIcon } from '../styledComponens';
import { PlayIcon, PauseIcon } from '../svg-icons';

const Root = styled('div')`
  ${baseIcon};
  margin-right: 20px;
`;

type Props = {
  typeActionPlayer: 'play' | 'pause',
  onClick?: () => void,
};

export const PlayPauseControl = ({ typeActionPlayer, onClick }: Props) => (
  <Root onClick={onClick}>{typeActionPlayer === 'pause' ? <PlayIcon /> : <PauseIcon />}</Root>
);

PlayPauseControl.defaultProps = {
  onClick: () => {},
};

export default withConnect(['typeActionPlayer'], { onClick: togglePause })(PlayPauseControl);
