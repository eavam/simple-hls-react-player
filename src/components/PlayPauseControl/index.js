import React from 'react'
import PropTypes from 'prop-types'
import { Icon } from '../../globalStyled.js'
import { PlayIcon, PauseIcon } from '../svg-icons'

const PlayPauseControl = ({ isPause, onClick }) =>
  <Root onClick={onClick}>
    {isPause ? <PlayIcon /> : <PauseIcon />}
  </Root>

PlayPauseControl.propTypes = {
  isPause: PropTypes.bool,
  onClick: PropTypes.func
}

const Root = Icon.extend`
  margin-right: 20px;
`

export default PlayPauseControl
