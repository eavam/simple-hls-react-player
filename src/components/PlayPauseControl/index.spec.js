import React from 'react'
import renderer from 'react-test-renderer'
import PlayPauseControl from './index.js'

test('PlayPauseControl', () => {

  const props = {
    isPause: false,
    onClick: () => {}
  }


  const component = renderer.create(
    <PlayPauseControl {...props} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.isFullscreen = true
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})