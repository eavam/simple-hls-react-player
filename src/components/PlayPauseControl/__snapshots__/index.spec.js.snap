// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayPauseControl 1`] = `
<div
  className="sc-bwzfXH knKTcO"
  onClick={[Function]}
>
  <svg
    className={undefined}
    viewBox="0 0 20.03 25"
  >
    <path
      d="M7 21.52a3.48 3.48 0 0 1-7 0v-18a3.48 3.48 0 1 1 7 0zm0 0M20 21.52a3.48 3.48 0 0 1-7 0v-18a3.48 3.48 0 0 1 7 0zm0 0"
      fill="#fff"
    />
  </svg>
</div>
`;

exports[`PlayPauseControl 2`] = `
<div
  className="sc-bwzfXH knKTcO"
  onClick={[Function]}
>
  <svg
    className={undefined}
    viewBox="0 0 20.03 25"
  >
    <path
      d="M7 21.52a3.48 3.48 0 0 1-7 0v-18a3.48 3.48 0 1 1 7 0zm0 0M20 21.52a3.48 3.48 0 0 1-7 0v-18a3.48 3.48 0 0 1 7 0zm0 0"
      fill="#fff"
    />
  </svg>
</div>
`;
